File: PascalTriangle/pascal_triangle.py
'''
Problem:
Print the N-th row of Pascal's triangle, e.g.

row 0: 1
row 1: 1 1
row 2: 1 2 1
row 3: 1 3 3 1
row 4: 1 4 6 4 1
row 5: 1 5 10 10 5 1

Solution:
Below is a program in Python 3 that solves the problem.
'''


def get_Nth_line_Pascal_triangle(N):
    if N == 0:
        res = [1]
    else:
        L = [1, 1]
        for k in range(N-1):
            L_new = [1]
            for i in range(len(L)-1):
                L_new.append(L[i] + L[i+1])
            L_new.append(1)
            L = L_new
        res = L
    string_list = [str(c) for c in res]
    return " ".join(string_list)


def task():
    # Example: Print rows 0 to 9 of Pascal's triangle
    for i in range(10):
        print("row " + str(i) + ": " + get_Nth_line_Pascal_triangle(i))


if __name__ == '__main__':
    task()

File: PascalTriangle/pascal_triangle.cpp
/*

Problem:
Print the N-th row of Pascal's triangle, e.g.

row 0: 1
row 1: 1 1
row 2: 1 2 1
row 3: 1 3 3 1
row 4: 1 4 6 4 1
row 5: 1 5 10 10 5 1

Solution:
Below is a program in C++11 that solves the problem.

Usage:
Compile with 
g++ -std=c++11 pascal_triangle.cpp -o pascal_triangle.out

and run the executable with
./pascal_triangle.out

*/


#include <bits/stdc++.h>
using namespace std;


string get_Nth_row_Pascal_triangle(int N) {
    vector<int> res;
    if (N == 0) {
        res.push_back(0);
    }
    else {
        vector<int> L;
        L.push_back(1);
        L.push_back(1);
        for (int i = 0; i < N-1; i++) {
            vector<int> L_new;
            L_new.push_back(1);
            for (int k = 0; k < L.size()-1; k++) {
                L_new.push_back(L[k] + L[k+1]);
            }
            L_new.push_back(1);
            L = move(L_new);
        }
        res = L;
    }
    string answer;
    for (auto &num : res) {
        answer.append(to_string(num));
        answer.append(" ");
    }
    return answer;
    
}

int main() {
    for (int i = 0; i < 10; i++) {
        cout << "row " << i << ": " << get_Nth_row_Pascal_triangle(i) << "\n";
    }          
    return 0;
}
